File: Python/ByRound/2051/2051_A_Preparing_for_the_Olympiad.py
# Problem: CF 2051 A - Preparing for the Olympiad
# https://codeforces.com/contest/2051/problem/A

"""
Problem: A. Preparing for the Olympiad

Purpose:
  Maximizes the difference (m - s) where m is the number of problems Monocarp solves,
  and s is the number of problems Stereocarp solves.
  
  Key insight:
    - Monocarp chooses which days to train.
    - If Monocarp trains on day i, then Stereocarp trains on day i+1 (if exists).
    - To maximize difference, for each day i, we can choose whether Monocarp trains or not.
      If Monocarp trains on day i, he gains a[i], but Stereocarp must train on day i+1 and loses b[i+1].
      
  Strategy:
    - For each day from 0 to n-2:
        If Monocarp trains on day i (a[i]), and Stereocarp trains on day i+1 (b[i+1]),
        we can compute the net gain of training on this day as (a[i] - b[i+1]).
    - If Monocarp trains on last day, he gets a[-1], since Stereocarp has no next day.

  Time Complexity: O(n) per test case
  Space Complexity: O(1) additional space (excluding input)
"""

for _ in range(int(input())):  # Read number of test cases
    t = int(input())  # Number of days
    c = 0  # Initialize counter for total difference
    a = [*map(int, input().split())]  # Problems Monocarp solves per day
    b = [*map(int, input().split())]  # Problems Stereocarp solves per day
    
    # Loop through all days except the last one
    for i in range(t - 1):
        # If Monocarp trains on day i and Stereocarp trains on day i+1,
        # the net gain for Monocarp is a[i] - b[i+1]
        if 0 < a[i] - b[i + 1]:
            c += a[i] - b[i + 1]
    
    # Add the last day's problem count (Monocarp trains on last day, no follow-up)
    print(c + a[-1])


# https://github.com/VaHiX/codeForces/

File: CPP/ByRound/2051/2051_A_Preparing_for_the_Olympiad.cpp
// Problem: CF 2051 A - Preparing for the Olympiad
// https://codeforces.com/contest/2051/problem/A

/*
A. Preparing for the Olympiad
Algorithm: Greedy
Time Complexity: O(n) per test case
Space Complexity: O(n) per test case

Monocarp wants to maximize the difference (m - s), where m is the number of problems he solves,
and s is the number of problems Stereocarp solves. 

Key insight:
- Monocarp can choose any subset of days to train.
- If Monocarp trains on day i (i < n), then Stereocarp trains on day i+1.
- On the last day, Stereocarp does not train.

Strategy:
We consider all possible ways for Monocarp to select training days and compute
the difference in total problems solved. 

The optimal strategy is to always include the last day in Monocarp's schedule,
as that maximizes his count (since he doesn't need to worry about the next day).

For each day i from 0 to n-2, if the difference between Monocarp's problems on day i
and Stereocarp’s problems on day i+1 is positive, we add it to the result. This simulates
the fact that in an optimal solution, we may want to avoid certain overlaps.

The first term is always a[n-1], because Monocarp should train on the last day.
Then, for every other day where Monocarp trains, we compare how much more he solves
than Stereocarp does on the following day.

*/

#include <cstdio>
#include <vector>
int main() {
  long t;
  scanf("%ld", &t);
  while (t--) {
    long n;
    scanf("%ld", &n);
    std::vector<long> a(n);
    for (long p = 0; p < n; p++) {
      scanf("%ld", &a[p]);
    }
    std::vector<long> b(n);
    for (long p = 0; p < n; p++) {
      scanf("%ld", &b[p]);
    }
    long res(a.back()); // Start with last day's Monocarp problems since that's always optimal
    for (long p = 0; p + 1 < n; p++) {
      long diff = (a[p] - b[p + 1]); // Difference on day p vs Stereocarp's day p+1
      res += (diff > 0 ? diff : 0);  // Only consider positive differences
    }
    printf("%ld\n", res);
  }
}


// https://github.com/VaHiX/codeForces/